frontend/src/components/tabs/MetricsTab.test.tsx

```diff
@@ -133,59 +188,102 @@ describe('MetricsTab', () => {
 
     await waitFor(() => getByText("Error in metrics artifact's confidenceMetrics data format."));
   });
+});
 
-  it('has no metrics', async () => {
-    jest.spyOn(mlmdUtils, 'getOutputArtifactsInExecution').mockReturnValue(Promise.resolve([]));
-    jest.spyOn(mlmdUtils, 'getArtifactTypes').mockReturnValue(Promise.resolve([]));
+describe('MetricsTab with confusionMatrix', () => {
+  it('shows confusion matrix', async () => {
     const execution = buildBasicExecution().setLastKnownState(Execution.State.COMPLETE);
-
+    const artifactType = buildBasicArtifactType();
+    const artifact = buildBasicArtifact();
+    artifact.getCustomPropertiesMap().set('name', new Value().setStringValue('metrics'));
+    artifact.getCustomPropertiesMap().set(
+      'confusionMatrix',
+      new Value().setStructValue(
+        Struct.fromJavaScript({
+          struct: {
+            annotationSpecs: [
+              { displayName: 'Setosa' },
+              { displayName: 'Versicolour' },
+              { displayName: 'Virginica' },
+            ],
+            rows: [{ row: [31, 0, 0] }, { row: [1, 8, 12] }, { row: [0, 0, 23] }],
+          },
+        }),
+      ),
+    );
+    jest.spyOn(mlmdUtils, 'getOutputArtifactsInExecution').mockResolvedValueOnce([artifact]);
+    jest.spyOn(mlmdUtils, 'getArtifactTypes').mockResolvedValueOnce([artifactType]);
     const { getByText } = render(
       <CommonTestWrapper>
         <MetricsTab execution={execution}></MetricsTab>
       </CommonTestWrapper>,
     );
-
-    await waitFor(() => getByText('There is no metrics artifact available in this step.'));
+    getByText('Metrics is loading.');
+    // We should upgrade react-scripts for capability to use libraries normally:
+    // https://github.com/testing-library/dom-testing-library/issues/477
```

Comment:
Shall we have a try?

Reply:
We should! I am more than happy to give it a try. But will make sure we complete V2 compatible first then we come to this, created issue: https://github.com/kubeflow/pipelines/issues/5794.